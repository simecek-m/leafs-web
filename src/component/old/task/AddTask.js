import React from "react";
import { withTranslation } from "react-i18next";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faPlus } from "@fortawesome/free-solid-svg-icons";
import "component/old/task/AddTask.sass";

export class AddTask extends React.Component {
  state = {
    active: false
  };

  constructor(props) {
    super(props);
    this.taskInput = React.createRef();
  }

  addTask(text) {
    const { onAdd } = this.props;
    onAdd(text);
    this.setState({
      active: false
    });
  }

  keyDown(keyCode) {
    switch (keyCode) {
      case 27:
        this.setState({
          active: false
        });
        break;
      case 13:
        this.addTask(this.taskInput.current.value);
        break;
      default:
        break;
    }
  }

  render() {
    const { t } = this.props;
    const addTaskContent = this.state.active ? (
      <input
        className="add-task-input"
        autoFocus
        ref={this.taskInput}
        onBlur={e => this.addTask(e.target.value)}
        onKeyDown={e => this.keyDown(e.keyCode)}
      />
    ) : (
      <span
        className="add-task-text"
        onClick={() => this.setState({ active: true })}
      >
        {t("addTask.new")}
      </span>
    );
    return (
      <div className="add-task-component">
        <FontAwesomeIcon className="icon" icon={faPlus} />
        {addTaskContent}
      </div>
    );
  }
}

export default withTranslation()(AddTask);
