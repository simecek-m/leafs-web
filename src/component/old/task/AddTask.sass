@import "colors.sass"

.add-task-component
  color: $text-secondary
  margin: 30px 0
  font-style: italic
  font-weight: bold
  font-size: 20px
  cursor: pointer
  transition: 0.5s ease all
  user-select: none

  &:hover
    color: $text-primary

    .add-task-input
      color: $text-primary

  .add-task-input
    background: none
    color: $text-secondary
    font-size: 20px
    border: none
    outline: none

  .icon
    margin-right: 10px
