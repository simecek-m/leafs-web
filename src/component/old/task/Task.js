import React from "react";
import { faCheckCircle } from "@fortawesome/free-regular-svg-icons";
import { faCheckCircle as SolidFaCheckCircle } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTrashAlt } from "@fortawesome/free-regular-svg-icons";
import "component/old/task/Task.sass";

function Task({
  text = "Task",
  completed = false,
  onClick = null,
  onDelete = null
}) {
  return completed ? (
    <div className="task-component completed" onClick={onClick}>
      <FontAwesomeIcon className="icon" icon={faCheckCircle} />
      {text}
      <FontAwesomeIcon
        className="icon-trash"
        icon={faTrashAlt}
        onClick={onDelete}
      />
    </div>
  ) : (
    <div className="task-component incomplete" onClick={onClick}>
      <FontAwesomeIcon className="icon" icon={SolidFaCheckCircle} />
      {text}
    </div>
  );
}

export default Task;
