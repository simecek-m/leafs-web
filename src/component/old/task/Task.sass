@import "colors.sass"

.task-component
  cursor: pointer
  font-size: 22px

  &:hover
    .icon-trash
      opacity: 1

  .icon
    width: 20px
    height: 20px
    margin-right: 10px
    transition: 0.5s ease all

  .icon-trash
    opacity: 0
    margin-left: 10px
    transition: 0.5s ease all

  &.incomplete
    margin-bottom: 10px

    &:hover
      .icon
        opacity: 1

    .icon
      opacity: 0

  &.completed
    font-weight: normal
    color: $text-secondary
