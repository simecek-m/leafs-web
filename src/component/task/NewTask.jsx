import { useKeyPress } from "hooks/interaction";
import { useEffect, useState } from "react";
import styles from "component/task/NewTask.module.sass";

export default function NewTask({ onCreate = () => null, onFocus, onBlur }) {
  const [text, setText] = useState("");

  const createNewTask = () => {
    onCreate({ text, completed: false });
    setText("");
  };

  const { addOnKeyDownEvent, removeOnKeyDownEvent } = useKeyPress(
    "Enter",
    createNewTask
  );

  useEffect(() => {
    addOnKeyDownEvent();
    return () => {
      removeOnKeyDownEvent();
    };
  }, [addOnKeyDownEvent, removeOnKeyDownEvent]);

  return (
    <input
      id={styles["new-task"]}
      onFocus={onFocus}
      onBlur={onBlur}
      value={text}
      onChange={(e) => setText(e.target.value)}
      placeholder="What do you need to remind?"
      autoComplete="off"
    />
  );
}
