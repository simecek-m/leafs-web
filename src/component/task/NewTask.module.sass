input#new-task
  outline: none
  border: none
  color: var(--on-background)
  font-size: 1.2rem
  width: 100%
  border-radius: 10px
  margin: 5px 0
  background: var(--surface)
  padding: 10px 20px
  box-sizing: border-box

  &::placeholder
    font-style: italic
    font-size: 0.8em
