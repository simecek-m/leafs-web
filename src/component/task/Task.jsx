import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styles from "component/task/Task.module.sass";
import { useState } from "react";

export default function Task({
  completed,
  id,
  text,
  onDelete,
  onStateChanged = () => null,
}) {
  const [checked, setChecked] = useState(completed);

  return (
    <span className={styles.checkbox}>
      <label htmlFor={id}>
        <input
          type="checkbox"
          id={id}
          checked={checked}
          onChange={() => {
            onStateChanged(!checked);
            setChecked(!checked);
          }}
        />
        <span>{text}</span>
      </label>
      <FontAwesomeIcon
        icon="close"
        className={styles["close-icon"]}
        onClick={() => onDelete(id)}
      />
    </span>
  );
}
