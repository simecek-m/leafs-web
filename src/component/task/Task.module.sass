span.checkbox
  display: flex
  align-items: center
  gap: 5px

  label
    font-size: 1.2em
    font-weight: bold

  input[type="checkbox"]
    appearance: none
    margin: 0
    font: inherit
    color: black
    width: 1.5em
    height: 1.5em
    border: 0.2em solid var(--on-background)
    display: grid
    place-content: center
    border-radius: 0.4em

    &::before
      content: ""
      width: 0.65em
      height: 0.65em
      opacity: 0
      transition: 120ms transform ease-in-out
      box-shadow: inset 1em 1em var(--background)
      transform-origin: bottom left
      clip-path: polygon(14% 44%, 0 65%, 50% 100%, 100% 16%, 80% 0%, 43% 62%)

    &:checked
      background: var(--gray)
      border-color: var(--gray)

      + label
        color: var(--gray)
        text-decoration: line-through

      &::before
        opacity: 1
