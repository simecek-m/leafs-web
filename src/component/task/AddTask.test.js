import React from "react";
import { shallow } from "enzyme";
import { AddTask } from "component/task/AddTask";

const TASK_STATE_NOT_ACTIVE = {
  active: false
};

const TASK_STATE_ACTIVE = {
  active: true
};

jest.mock("react-i18next", () => ({
  withTranslation: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: key => key };
    return Component;
  }
}));

describe("AddTask component", () => {
  test("should render default component", () => {
    const wrapper = shallow(<AddTask />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should switch state to active", () => {
    const wrapper = shallow(<AddTask />);
    let state = wrapper.state();
    expect(wrapper).toMatchSnapshot();
    expect(state).toEqual(TASK_STATE_NOT_ACTIVE);
    wrapper.find(".add-task-text").simulate("click");
    state = wrapper.state();
    expect(state).toEqual(TASK_STATE_ACTIVE);
    expect(wrapper).toMatchSnapshot();
  });
});
