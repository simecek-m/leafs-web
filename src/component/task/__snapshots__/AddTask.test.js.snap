// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTask component should render default component 1`] = `
<div
  className="add-task-component"
>
  <FontAwesomeIcon
    className="icon"
    icon={
      Object {
        "icon": Array [
          448,
          512,
          Array [],
          "f067",
          "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
        ],
        "iconName": "plus",
        "prefix": "fas",
      }
    }
  />
  <span
    className="add-task-text"
    onClick={[Function]}
  >
    addTask.new
  </span>
</div>
`;

exports[`AddTask component should switch state to active 1`] = `
<div
  className="add-task-component"
>
  <FontAwesomeIcon
    className="icon"
    icon={
      Object {
        "icon": Array [
          448,
          512,
          Array [],
          "f067",
          "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
        ],
        "iconName": "plus",
        "prefix": "fas",
      }
    }
  />
  <span
    className="add-task-text"
    onClick={[Function]}
  >
    addTask.new
  </span>
</div>
`;

exports[`AddTask component should switch state to active 2`] = `
<div
  className="add-task-component"
>
  <FontAwesomeIcon
    className="icon"
    icon={
      Object {
        "icon": Array [
          448,
          512,
          Array [],
          "f067",
          "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
        ],
        "iconName": "plus",
        "prefix": "fas",
      }
    }
  />
  <input
    autoFocus={true}
    className="add-task-input"
    onBlur={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;
